common/src/main/java/org/apache/falcon/entity/parser/DatasourceEntityParser.java

```diff
@@ -51,6 +51,10 @@ public DatasourceEntityParser() {
     public void validate(Datasource db) throws FalconException {
         try {
             ClassLoader hdfsClassLoader = HdfsClassLoader.load(db.getName(), db.getDriver().getJars());
+            DatasourceHelper.validateCredential(DatasourceHelper.getCredential(db));
```

Comment:
not sure if I understand it correctly. In that case, we would need a method like validateCredentialHelper(Credential)? this is because the way the xsd provides credentials for each interface.